docs/src/content/docs/utilities/add-virtual-import.mdx

```diff
@@ -0,0 +1,58 @@
+---
+title: addVirtualImport
+description: Allows adding Vite Virtual Modules easily to access data across your integration.
+---
+
+`addVirtualImport` allows you to create modules and store state across your integration. For example:
+
+```ts title="integration/index.ts"
+import type { AstroIntegration } from "astro";
+import { addVirtualImport } from "astro-integration-kit";
+
+export default function myIntegration(): AstroIntegration {
+    return {
+        name: "my-integration",
+        hooks: {
+            "astro:config:setup": ({ updateConfig }) => {
+                addVirtualImport({
+                    name: 'virtual:my-integration/config', // this will now be importable with `virtual:my-integration/config`
+                    content: `export default ${ JSON.stringify({foo: 'bar'}) }`,
+                    updateConfig,
+                })
+            }
+        }
+    }
+}
+```
+
+So to read this config anywhere else in your integration is easy:
+
+```ts title="my-integration/src/pages/index.astro"
+---
+import config from 'virtual:my-integration/config',
+
+console.log(config.foo) // 'bar'
+---
+```
+
+Now you can easily share state across your integration 🎉!
+
+:::caution
+Virtual Modules defined with `addVirtualImport` cannot be referenced inside the `astro-integration-kit` integration utils.
+:::
+
+## Creating a `virtual.d.ts`
+
+In order to get the types for your virtual modules you'll need to manually declare a virtual module.
+
+First create a `virtual.d.ts` file in your integration root.
+
+Then you can define your virtual modules as global TypeScript modules:
+
+```ts title="my-integration/virtual.d.ts"
+declare module 'virtual:astro-integration-kit-playground/config' {
+    export default { foo: string };
+}
+```
+
+Now you'll have perfect typing when importing your virtual modules 👏!
```

Comment:
Can you add a note below explaining that those types are not accessible to the end users?